### PR TITLE
Drop support for SQLite

### PR DESCRIPTION
**Pull request by rclough**
After some private discussion, regarding the future of dagobah features, etc, it seems that supporting SQLite, at least for now, is a severe hinderance to development time, with little benefit.

* Making data model changes, even minor, often involve annoying alembic migrations
* Making data model changes on such a small, relatively early project, often involves serious rearchitecting of SQL tables, often bending over backwards to support things that mongo practically does baked in
* There's really only 2 main developers in the project (who are not super active either), neither of which enjoy the SQL aspect of dev
* SQLite offers little advantage as a back end. 
    * Nobody should be running potentially critical systems on something like sqlite (except maybe in embedded applications)
    * SQLite gets very slow very fast (especially considering logging)
* Setting up mongo is fairly trivial on most systems (and dagobah runs fine with stock mongo configs), and if you're testing, you can run it without a back end.

Note that this action does not mean we actively discourage the use of RDBMS or alternate back ends, in act the back end is left as to still be agnostic, so if someone wanted to fork or contribute new back ends to dagobah, they are plenty welcome!

### CODE REVIEW

File: dagobah/backend/base.py [hunk 1]
```diff
@@ -41,12 +40,7 @@ def verify_required_packages(self):
                                                                                                         spec['version']))
                 continue
 
-            # everyone has to be special
-            if spec['pypi_name'] == 'pysqlite':
-                module = __import__('pysqlite2._sqlite')
-                installed_version = getattr(module._sqlite, spec['version_key'])
-            else:
-                installed_version = getattr(module, spec['version_key'])
+            installed_version = getattr(module, spec['version_key'])
```
Comment (rclough) [inline dagobah/backend/base.py]: I did this part somewhat blindly, but I think this is correct now that SQLite is out of the picture?

File: dagobah/daemon/dagobahd.yml [hunk 2]
```diff
@@ -17,9 +17,8 @@ Dagobahd:
 
   # choose one of the available backends
   # None: Dagobah will not use a backend to permanently store data
-  # sqlite: use a SQLite database. see the SQLite section in this file
   # mongo: store results in MongoDB. see the MongoBackend section in this file
-  backend: sqlite
+  backend: mongo
```
Comment (rclough) [inline dagobah/daemon/dagobahd.yml]: not ure if we should default to mongo or None

File: dagobah/backend/base.py (hunk 1 above)
Comment (thieman) [inline dagobah/backend/base.py]: Yeah, just needed to remove the hack for sqlite

File: dagobah/daemon/dagobahd.yml (hunk 2 above)
Comment (thieman) [inline dagobah/daemon/dagobahd.yml]: Probably `None` so people can immediately get up and running to play around with it

**Comment by thieman** [comment]
This seems legit. As discussed, I'll do the following, probably sometime this weekend.

1. Cut a release
1. Make sure this is all good and merge it
1. Cut another release with this in it

**Comment by danielerapati** [comment]
Ouch: I use sqlite as a back-end and I was looking forward to JobTask. 
Perhaps I can branch the next release and try to back-port support for JobTask on sqlite once it is up and running on Mongo?

**Comment by rclough** [comment]
You're more than welcome to do that if you'd like!


Sent from my Verizon Wireless 4G LTE smartphone


-------- Original message --------

**Comment by rclough** [comment]
@danielerapati out of curiosity, why do use sqlite? Do you not experience performance issues?

**Comment by thieman** [comment]
How difficult would it be to leave the SQLite backend intact but add a feature tagging system so that JobTask is only available to people using a supported backend? I'm skeptical anyone would ever be able to implement the entire SQL backend again from scratch, but implementing one feature at a time might be doable.

**Comment by rclough** [comment]
I've no idea, I've never done such a thing, but I am not opposed to it.

**Comment by rclough** [comment]
I'm more than happy to let someone else tackle the sql backend issue of supporting job-in-job stuff, as long as it doesnt get too in the way of my regular development. I for one don't want to think too much about SQL beyond basic ORM type stuff

**Comment by thieman** [comment]
@rclough Think it'd make sense for you to keep working on JobTasks and I'll either come up with a way to make feature tagging work or merge this PR.

**Comment by rclough** [comment]
Sounds good! Worth noting though that mongo-only job-in-job data model support is only really a day away from a PR probably (just deserialization code and stuff). Although now that I say that, its just back end so its not like anyone using SQL would see the disparity

**Comment by danielerapati** [comment]
@rclough performance is quite horrible but my jobs are not very frequent
(daily, bi-daily at worst)
 and I rarely make changes (say once a week) so I barely notice.
I started out with sqlite when I was evaluating dagobah (lazyness), did not
encounter any issue right away and after a couple of months I was quite
locked into it, I even ended up source controlling the sqlite .db file.

**Comment by thieman** [comment]
@rclough I haven't been able to work on Dagobah since May and I don't see that changing in the near future. Without anyone stepping up to work on a feature tagging system, I'd like to merge this now. Any objections?

**Comment by rclough** [comment]
I'm down. I don't really have time to do feature tagging as it's not really a priority for me/us personally

**Pull request merged**